### DISCUSSION TITLE
Minimum object size for detection

### DISCUSSION THREAD

**Discussion by chmjelek** [ultralytics/yolov5#10880] [General]
What would be recommended minimal size of object (in pixels, relative to image size) for which detection is reasonable? I realize that it depends on many factors, but let's determine that they are constant (yolov5m6, conf_thr=0.25, etc).

Let's say the images have a fixed resolution (1400x1800 pixels); and the task is to detect the people. How would you determine the smallest acceptable object size for detection to be possible? Is there any rule of thumb?

**Comment by evia1** [comment]
Wondering as well, because I'm getting this warning during autoanchor:

WARNNING ⚠️ Extremely small objects found: 73 of 12265 labels are <3 pixels in size

**Comment by wb666greene** [comment]
Can't speak for this yolo5, but I've been running MobilenetSSD_v2 with 4K (3840x2160 pixels) security cameras, asymmetrically resized to 300x300 pixels for person detection and when scaled back to the full 4k frame the bounding boxes are often  smaller than 150 pixels.  How small is too small depends on the cameras field of view.  As a practical matter I reject detections that aren't "tall enough"  because it is either not "real" or is outside of my property where I don't want notifications. I also reject detection boxes  that are too large as one of my major false positives is bugs detecting as a "person" when flying in front of the camera filling half or more of the field of view as an out of focus blob.

My starting point for cameras that have an "across the street" view is that a bounding box height smaller than ~5% of the frame height when scaled for the original frame size  is probably bogus or too far away to matter.
